### ISSUE TITLE
__doc__ is reported useless by flake8

### ISSUE THREAD

**Issue by bcm-at-zama** [PyCQA/flake8#1589]
### how did you install flake8?

```console
poetry
```

### unmodified output of `flake8 --bug-report`

```json
{
  "dependencies": [],
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.9.12",
    "system": "Darwin"
  },
  "plugins": [
    {
      "is_local": false,
      "plugin": "flake8-bugbear",
      "version": "22.3.23"
    },
    {
      "is_local": false,
      "plugin": "flake8-darglint",
      "version": "1.8.1"
    },
    {
      "is_local": false,
      "plugin": "mccabe",
      "version": "0.6.1"
    },
    {
      "is_local": false,
      "plugin": "pycodestyle",
      "version": "2.8.0"
    },
    {
      "is_local": false,
      "plugin": "pyflakes",
      "version": "2.4.0"
    }
  ],
  "version": "4.0.1"
}
```

### describe the problem

I got:

b.py:6:5: F841 local variable '__doc__' is assigned to but never used

But __doc__ is used eg for Sphinx

#### what I expected to happen

no error

#### sample code

```python
"""Test"""
import numpy


def f():
    __doc__ = numpy.ones.__doc__


if __name__ == "__main__":
    f()
```

#### commands ran

```console
$ flake8 b.py
...
```

**Comment by asottile** [comment]
your assignment indeed does nothing, the linter is not wrong

but also you did not read the issue template and have reported this in the wrong place

**Comment by bcm-at-zama** [comment]
Thanks
